movement/watch_faces/complication/breathing_face.c

```diff
@@ -0,0 +1,198 @@
+    }
+}
+
+void breathing_face_activate(movement_settings_t *settings, void *context) {
+    // same as above: silence the warning, we don't need to check the settings.
+    (void) settings;
+    // we do however need to set some things in our context. Here we cast it to the correct type...
+    breathing_state_t *state = (breathing_state_t *)context;
+    // ...and set the initial state of our watch face.
+    state->current_stage = 0;
+    state->sound_on = true;
+}
+
+const int NOTE_LENGTH = 80;
+
+void beep_in (void) {
+        const BuzzerNote notes[] = {
+            BUZZER_NOTE_C4,
+            BUZZER_NOTE_D4,
+            BUZZER_NOTE_E4,
+        };
+        const uint16_t durations[] = {
+            NOTE_LENGTH,
+            NOTE_LENGTH,
+            NOTE_LENGTH
+        };
+        for(size_t i = 0, count = sizeof(notes) / sizeof(notes[0]); i < count; i++) {
+            watch_buzzer_play_note(notes[i], durations[i]);
+        }
+}
+
+void beep_in_hold (void) {
+        const BuzzerNote notes[] = {
+            BUZZER_NOTE_E4,
+            BUZZER_NOTE_REST,
+            BUZZER_NOTE_E4,
+        };
+        const uint16_t durations[] = {
+            NOTE_LENGTH,
+            NOTE_LENGTH * 2,
+            NOTE_LENGTH,
+        };
+        for(size_t i = 0, count = sizeof(notes) / sizeof(notes[0]); i < count; i++) {
+            watch_buzzer_play_note(notes[i], durations[i]);
+        }
+}
+
+void beep_out (void) {
+        const BuzzerNote notes[] = {
+            BUZZER_NOTE_E4,
+            BUZZER_NOTE_D4,
+            BUZZER_NOTE_C4,
+        };
+        const uint16_t durations[] = {
+            NOTE_LENGTH,
+            NOTE_LENGTH,
+            NOTE_LENGTH,
+        };
+        for(size_t i = 0, count = sizeof(notes) / sizeof(notes[0]); i < count; i++) {
+            watch_buzzer_play_note(notes[i], durations[i]);
+        }
+}
+
+void beep_out_hold (void) {
+        const BuzzerNote notes[] = {
+            BUZZER_NOTE_C4,
+            BUZZER_NOTE_REST * 2,
+            BUZZER_NOTE_C4,
+        };
+        const uint16_t durations[] = {
+            NOTE_LENGTH,
+            NOTE_LENGTH,
+            NOTE_LENGTH,
+        };
+        for(size_t i = 0, count = sizeof(notes) / sizeof(notes[0]); i < count; i++) {
+            watch_buzzer_play_note(notes[i], durations[i]);
+        }
+}
+
+bool breathing_face_loop(movement_event_t event, movement_settings_t *settings, void *context) {
+    (void) settings;
+    breathing_state_t *state = (breathing_state_t *)context;
+
+    switch (event.event_type) {
+        case EVENT_ACTIVATE:
+        case EVENT_TICK:
+
+            if (state->sound_on == true) {
+                watch_set_indicator(WATCH_INDICATOR_BELL); 
+            } else {
+                watch_clear_indicator(WATCH_INDICATOR_BELL); 
+            }
+
+            switch (state->current_stage)
+            {
+            case 0: { watch_display_string("Breath", 4); if (state->sound_on) beep_in(); } break;
```

Comment:
It's fine like this, but I would be tempted to put this data in an array. In fact, I think you could represent the whole set of steps as something like:

```
struct {
    char *str;
    void (*sound_fn)(void);
} stages[] = {
    {"Breath", beep_in},
    {"In    3", noop},
    ...
};
```

But I'm probably just saying this because I don't like switch statements that much :)

Reply:
I like this. I actually wanted remove the data from the beeping portion, too. But that's where my non-existing proficiency in C stopped me. No completely sure where I need the pointers and how to deal with arrays in that context. Still need to learn that. Thanks.